src/libs/antares/study/scenario-builder/TSnumberData.cpp

```diff
@@ -560,13 +560,23 @@ void ntcTSNumberData::saveToINIFile(const Study& study, Yuni::IO::File::Stream&
     for (const auto& i : pArea->links)
     {
         const Data::AreaLink* link = i.second;
+        /*
```

Comment:
As you know, this is a workaround. 
This added block may not belong here, because : 
- it tries to do a job that should have been done before in the execution,
- it makes the current task less clear than it should.

Isn't it possible to make that job (for instance) in the **AreaList::renameArea(...)** function ?
If it's not, we should consider a refactoring to make it possible, I think.

Reply:
You're right. I think lines 569-572 are not mandatory, if we check that areas have already been renamed when saving the temporary file sc-builder data.


The order is the following when renaming an area :
1. Rename areas
2. Save the sc. builder data in a temporary file
3. Change data in memory
4. Load the sc. builder data from the temporary file

However, we should not use `i.first` there is no way map `Area::links` is up-to-date at step 1, since it is updated (and must be updated) at step 3,